util/metrics/kubectl/kubectl_metrics_test.go

```diff
@@ -0,0 +1,11 @@
+package kubectl
+
+import (
+	"testing"
+)
+
+func Test_RegisterWithClientGo_race(_ *testing.T) {
+	// This test ensures that the RegisterWithClientGo function can be called concurrently without causing a data race.
+	go RegisterWithClientGo()
+	go RegisterWithClientGo()
```

Comment:
As the test doesn't have any assertion, the 2 goroutines will be terminated as soon as it triggers the second one. Maybe we should somehow add an assertion to make sure both goroutines are executed to the end.